client/src/Actions/userSlice.ts

```diff
@@ -0,0 +1,66 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export type UserType = {
```

Comment:
Może zrobic to jako rozszerzenie SignupUserType, będzie mniej kodu i przejrzyściej bo to są te same propsy :D

Reply:
ok a jak mamy reject z requesta to wtedy musimy do usera przypisac wartość null, bo  go  nie ma i wtedy musi być type null. Chyba że masz jakiś pomysł jak  to rozwiązć, bo ja nw ? :)